### ISSUE TITLE
[feature request] display space ID for every active monitor

### ISSUE THREAD

**Issue by DoDoENT** [gechr/WhichSpace#24]
Would it be possible to display a space ID for each monitor?

**Comment by marcelniessen** [comment]
This would be a nice feature. Currently there is shown the same space ID across all monitors. The space ID shown is currently always the last one changed.

e.g. if I switch from space 1 to space 2 on monitor 1, monitor 2 (which is displaying space 3) also shows space id 2 in the menu bar.

**Comment by KingOfSpades** [comment]
Would love this as wel!

**Comment by ksalzke** [comment]
This would be fantastic if it's possible!

**Comment by DerWeiseScheuch** [comment]
Yes! I have a 3 Monitor desk using 8 spaces (1-8) + 4 spaces (9-12) + 4 spaces (13-16) and one number on all screens is pretty useless for my needs.

**Comment by alexus37** [comment]
Forked the repo and created a working prototype, which can be used.
[Check out here](https://github.com/alexus37/WhichSpace)
However, because of my non-existent swift skills, I think someone else would be more suitable in making a proper pr.

 Here an example for 3 screens.
![WhichSpace](https://user-images.githubusercontent.com/7422767/146915683-8fcb7994-15bc-4479-adb8-4e183dfd8ed3.png)

**Comment by TheophileMot** [comment]
@alexus37 That looks good, but it also seems like it takes up a lot of room. (Consider the case of @DerWeiseScheuch, where there are 16 spaces.) It might be nice to have an option to expand it like this or not.

**Comment by KingOfSpades** [comment]
Or we could use fractions like ¼ and ⅜ 

![CleanShot 2021-12-30 at 09 25 49](https://user-images.githubusercontent.com/2156115/147734538-0515eefc-daca-47d1-a92b-141c6786ba02.png)

But maybe that's a bit to small

**Comment by alexus37** [comment]
So for me, this is fine to take that much space, since I come from the I3 world I am used to this. Also, in combination with keyboard shortcuts, this is good information to have (for me at least). I guess the right approach would be to allow multiple visualizations, which the user can configure, i.e. full detail vs compact.

**Comment by KingOfSpades** [comment]
> Forked the repo and created a working prototype, which can be used. [Check out here](https://github.com/alexus37/WhichSpace) However, because of my non-existent swift skills, I think someone else would be more suitable in making a proper pr.
> 
> Here an example for 3 screens. ![WhichSpace](https://user-images.githubusercontent.com/7422767/146915683-8fcb7994-15bc-4479-adb8-4e183dfd8ed3.png)

Any chance of pre compiling it and adding the binary to Git?

**Comment by alexus37** [comment]
Yes, I can do that next week.

**Comment by AugustM** [comment]
I am currently using 21 Desktop Spaces, spread between my MacBook and an external monitor.

Using fraction symbols to display #/# would limit me to 8 desktops per monitor, not what I would need. Similarly, having all 21 numbers in the Toolbar, just to have two of them highlighted, would be a serious waste of toolbar space, either this display or other toolbar icons would be regulary eclipsed by the current app's menus.

What's wrong with 2, 3, or 4 instances of the current, single number display?

**Comment by alexus37** [comment]
I am using shortcuts eg. opt+<number> to switch between spaces therefore having more than 10 does not make sense to me. Additionally, I am also using shift+opt+<number> to move an application from one space to the other, therefore I want to know which spaces are on which screen and which one is visible. This works well for me, however, I don't want to state this is the perfect way.

**Comment by KingOfSpades** [comment]
So, I checked out https://github.com/Jaysce/Spaceman (which is a fork of Which space). It has the features I was looking .

**Comment by pascalpp** [comment]
@alexus37 

> Additionally, I am also using shift+opt+number to move an application from one space to the other

How do you do that?

**Comment by TheophileMot** [comment]
> How do you do that?

@pascalpp This is possible using (for example) [BetterTouchTool](https://folivora.ai/), which is an exceptionally useful app.

**Comment by alexus37** [comment]
I'm using yabai + skhd! In the config file, you have the power to define a command that moves the current window to another space. For example:

```
shift + lalt - 1 : yabai -m window --space 1
```

**Comment by gechr** [comment]
https://github.com/gechr/WhichSpace/releases/tag/v0.4.1 should now update the icon when you focus a new display, to represent the active space on that particular display.

**Comment by dafeder** [comment]
What I'm seeing is the same icon on both display, showing whatever's active. What I'd like to see is, on my inactive screen, the visible space number, even though it doesn't have the focus currently. So I might have a "1" on one monitor and a "4" on my second, and it doesn't change if I move from one monitor to another, just when I switch spaces on either one. Is that possible, maybe just with a different combination of settings? It's what I expected when I selected "separate icons per display" but not seeing that behavior.

**Comment by gechr** [comment]
@dafeder _"Separate icons per Display"_ is a slightly misleading description of what the option actually does. As you discovered, it doesn't show a distinct icon representing the active Space per display.

I've tried to get this feature to work in the past, because I personally wanted it too, but ultimately conceded it's a macOS limitation, i.e. a single application can't show different icons in the menu bar depending on the display.

If anyone can point me at an example application that does something similar, or some documentation, I'd be happy to try and implement it.

**Comment by dafeder** [comment]
Thanks for the reply, not a mac developer so can't really give you ideas. Can you just make an icon only appear on a single monitor? If so could maybe be accomplished with two instances of the same app?
